source/Nevermore/RelationalStore.cs

```diff
@@ -71,18 +87,25 @@ public class RelationalStore : IRelationalStore
             JsonSerializerSettings jsonSettings,
             IRelatedDocumentStore relatedDocumentStore,
             int keyBlockSize = 20,
-            ObjectInitialisationOptions objectInitialisationOptions = ObjectInitialisationOptions.None)
+            ObjectInitialisationOptions objectInitialisationOptions = ObjectInitialisationOptions.None,
+            bool setMarsAsDefault = true)
         {
             this.registry = new Lazy<RelationalTransactionRegistry>(
-                () => SetConnectionStringOptions(connectionString(), applicationName)
+                () => SetConnectionStringOptions(connectionString(), applicationName, false)
+            );
+
+            this.registryWithMarsOn = new Lazy<RelationalTransactionRegistry>(
+                () => SetConnectionStringOptions(connectionString(), applicationName, true)
             );
+
             this.sqlCommandFactory = sqlCommandFactory;
             this.mappings = mappings;
             keyAllocator = new KeyAllocator(this, keyBlockSize);
 
             this.jsonSettings = jsonSettings;
             this.relatedDocumentStore = relatedDocumentStore;
             this.objectInitialisationOptions = objectInitialisationOptions;
+            this.setMarsAsDefault = setMarsAsDefault;
         }
 
         public string ConnectionString => registry.Value.ConnectionString;
```

Comment:
I believe we will have to make the **Lazy registry** smarter as switching between different lazy factories is going to cause problems for code like the above `public string ConnectionString => registry.Value.ConnectionString;` as it depends on the **registry** not the other one. In other words, in the case of `registryWithMarsOn` in use, the above code and other code that depends on `registry` are no longer reflecting the actual registry instance that is in use any more. 

And I assume it's an expensive operation to initialize a `RelationalTransactionRegistry` instance (as we are using lazy initialization), initializing two instances may have some performance impact.

Reply:
> public string ConnectionString => registry.Value.ConnectionString;
👍 

> And I assume it's an expensive operation to initialize a RelationalTransactionRegistry instance (as we are using lazy initialization), initializing two instances may have some performance impact.

No, I don't think so.  It looks like we simply don't want to execute a delegate in the constructor.